src/components/main-app/ol-init.js

```diff
@@ -14,6 +14,12 @@ export default function OlInit() {
   const prevTiles = usePrevious(tiles);
   const prevShape = usePrevious(shape);
 
+  const autoZoom = (map, shapeSource) => {
+    setTimeout(() => {
+      map.getView().fit(shapeSource.getExtent())
+    }, 500)
+  }
+
```

Comment:
2 Things - 

1. There is an event function you can use to do it (no need to use setTimeout)
2. add `{ padding: [20, 20, 20, 420] }` in `fit()` (check how you can do this)... So that it comes to centre